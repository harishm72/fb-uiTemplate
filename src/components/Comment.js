import React from 'react';
import '../styles/App.css';

const Comment = ({comment, index }) => {
        return(
            <div key={index} className="comment">
                <p>{comment['comment']}</p>
                <p className="comment-time">Created at : {comment['created_at']}</p>
            </div>
        )
    }
export default Comment;