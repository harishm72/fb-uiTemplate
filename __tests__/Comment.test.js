import React from 'react';
import { shallow, mount } from 'enzyme';
import Comment from '../src/components/Comment';
import renderer from 'react-test-renderer'

let comment = {
    "comment": "Well said, i agree.",
    "created_at": "Fri May 11 2018 19:17:43 GMT+0530 (IST)"
};

let index = 0;

describe('>>>Comment --- Snapshot', () => {
    it('Rednders the Comment Component', () => {
        const renderedValue = renderer.create(<Comment key={index} comment={comment} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('>>>H O M E --- Shallow Render REACT COMPONENTS', () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallow(<Comment key={index} comment={comment} />)
    })

    it('render the Comment component', () => {
        expect(wrapper.length).toEqual(1)
    });

    it('has comment with tag p' , () => {
        expect(wrapper.contains(<p>{comment['comment']}</p>)).toBe(true)
    })
    
    it('has comment displaying comment text' , () => {
      expect(wrapper.find('p').get(0).props.children).toBe(comment['comment'])
    })

    it('has comment creation date-time with tag p' , () => {
        expect(wrapper.contains(<p className="comment-time">Created at : {comment['created_at']}</p>)).toBe(true)
    })
    
    it('has created_at displaying comment creation date-time' , () => {
      expect(wrapper.find('p').get(1).props.children).toEqual(["Created at : ", comment['created_at']])
    })
});