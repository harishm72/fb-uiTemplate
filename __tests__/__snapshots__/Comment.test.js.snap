// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Comment --- Snapshot Rednders the Comment Component 1`] = `
<div
  className="comment"
>
  <p>
    Well said, i agree.
  </p>
  <p
    className="comment-time"
  >
    Created at : 
    Fri May 11 2018 19:17:43 GMT+0530 (IST)
  </p>
</div>
`;
